### PR TITLE
Improve query performance of system tables

### PR DESCRIPTION
**Pull request by zhongyuankai**
Changelog category (leave one):
- Improvement

In a scenario with a particularly large amount of tables and data, querying the `system.tables` is very slow. The survey found that the acquisition of four fields is very time-consuming. For example `total_rows` , `total_bytes`, `create_table_query`, `engine_full`, my idea is to add an index to the `name` field to filter out unnecessary tables in advance.

```
bigdata-clickhouse-public00.ys :) select  * from system.tables where name = 'xxx';

SELECT *
FROM system.tables
WHERE name = 'xxx'

Ok.

0 rows in set. Elapsed: 22.154 sec. Processed 12.00 thousand rows, 44.54 MB (541.53 rows/s., 2.01 MB/s.)
```

### CODE REVIEW

File: src/Storages/System/StorageSystemTables.cpp [hunk 1]
```diff
@@ -88,6 +88,26 @@ static ColumnPtr getFilteredDatabases(const SelectQueryInfo & query_info, Contex
     return block.getByPosition(0).column;
 }
 
+static ColumnPtr getFilteredTables(const ASTPtr & query, ColumnPtr & filtered_databases_column, ContextPtr context)
+{
+    MutableColumnPtr column = ColumnString::create();
+
+    size_t database_idx = 0;
+    for (; database_idx < filtered_databases_column->size(); ++database_idx)
```
Comment (azat) [inline src/Storages/System/StorageSystemTables.cpp:96]:
```suggestion
    for (size_t database_idx = 0; database_idx < filtered_databases_column->size(); ++database_idx)
```

**Review by vdimir — APPROVED** [review]
Thank you for your changes!

**Comment by vdimir** [comment]
<details>
<summary>Test script</summary>

```bash
#!/bin/bash

set -eu

BATCHSIZE=16

for (( i = 0; i < ${1:-10000}; i++ )); do
	((i % BATCHSIZE == 0)) && wait && echo "Completed ${i}"
	clickhouse client --query "CREATE TABLE IF NOT EXISTS foo_${i} (x UInt32) ENGINE = Memory" 
done
wait && echo "CREATE completed"

clickhouse client -t --query "select count() from system.tables"

clickhouse client -t --query "select * from system.tables WHERE name == 'foo_123' FORMAT Null"

```

</details>

Last query takes `~0.5 sec` without patch and `~0.04 sec` on version from this PR, great

**Comment by zhongyuankai** [comment]
Thank you for your review, it's great.

**Comment by huzhichengdd** [comment]
It is awesome!

**Pull request merged**